### PR TITLE
v7k monitoring

### PR DESCRIPTION
**Pull request by brianhu2006**
add v7k monitoring.
1. add v7k status 
```
root@ds0055:~# sudo /etc/sensu/plugins/check-v7k-canister.py -v 10.254.66.43 -u bbcmonitor -i /etc/cinder/v7k/ssh/monitoring/id_rsa -p 22 --criticality critical
canister-status check OK:Both Canisters are online
```
2. add  v7k metrics 

```root@ds0055:~# sudo /etc/sensu/plugins/metrics_v7k.py -v 10.254.66.43 -u bbcmonitor -i /etc/cinder/v7k/ssh/monitoring/id_rsa  -p 22  --scheme stats.bb.colympus-lab.openstac
stats.bb.colympus-lab.openstac.node1.compression_cpu_pc	0	1494572996
stats.bb.colympus-lab.openstac.node1.cpu_pc	2	1494572996
stats.bb.colympus-lab.openstac.node1.fc_mb	0	1494572996
stats.bb.colympus-lab.openstac.node1.fc_io	0	1494572996
stats.bb.colympus-lab.openstac.node1.sas_mb	0	1494572996
stats.bb.colympus-lab.openstac.node1.sas_io	0	1494572996
stats.bb.colympus-lab.openstac.node1.iscsi_mb	0	1494572996
stats.bb.colympus-lab.openstac.node1.iscsi_io	0	1494572996
stats.bb.colympus-lab.openstac.node1.write_cache_pc	0	1494572996
stats.bb.colympus-lab.openstac.node1.total_cache_pc	0	1494572996
stats.bb.colympus-lab.openstac.node1.vdisk_mb	0	1494572996
stats.bb.colympus-lab.openstac.node1.vdisk_io	0	1494572996
stats.bb.colympus-lab.openstac.node1.vdisk_ms	0	1494572996
stats.bb.colympus-lab.openstac.node1.mdisk_mb	0	1494572996
stats.bb.colympus-lab.openstac.node1.mdisk_io	0	1494572996
stats.bb.colympus-lab.openstac.node1.mdisk_ms	0	1494572996
stats.bb.colympus-lab.openstac.node1.drive_mb	0	1494572996
stats.bb.colympus-lab.openstac.node1.drive_io	0	1494572996
stats.bb.colympus-lab.openstac.node1.drive_ms	0	1494572996
stats.bb.colympus-lab.openstac.node1.vdisk_r_mb	0	1494572996
stats.bb.colympus-lab.openstac.node1.vdisk_r_io	0	1494572996
stats.bb.colympus-lab.openstac.node1.vdisk_r_ms	0	1494572996
stats.bb.colympus-lab.openstac.node1.vdisk_w_mb	0	1494572996
stats.bb.colympus-lab.openstac.node1.vdisk_w_io	0	1494572996
stats.bb.colympus-lab.openstac.node1.vdisk_w_ms	0	1494572996
stats.bb.colympus-lab.openstac.node1.mdisk_r_mb	0	1494572996
stats.bb.colympus-lab.openstac.node1.mdisk_r_io	0	1494572996
stats.bb.colympus-lab.openstac.node1.mdisk_r_ms	0	1494572996
stats.bb.colympus-lab.openstac.node1.mdisk_w_mb	0	1494572996
stats.bb.colympus-lab.openstac.node1.mdisk_w_io	0	1494572996
stats.bb.colympus-lab.openstac.node1.mdisk_w_ms	0	1494572996
stats.bb.colympus-lab.openstac.node1.drive_r_mb	0	1494572996
stats.bb.colympus-lab.openstac.node1.drive_r_io	0	1494572996
stats.bb.colympus-lab.openstac.node1.drive_r_ms	0	1494572996
stats.bb.colympus-lab.openstac.node1.drive_w_mb	0	1494572996
stats.bb.colympus-lab.openstac.node1.drive_w_io	0	1494572996
stats.bb.colympus-lab.openstac.node1.drive_w_ms	0	1494572996
stats.bb.colympus-lab.openstac.node1.iplink_mb	0	1494572996
stats.bb.colympus-lab.openstac.node1.iplink_io	0	1494572996
stats.bb.colympus-lab.openstac.node1.iplink_comp_mb	0	1494572996
stats.bb.colympus-lab.openstac.node2.compression_cpu_pc	0	1494572996
stats.bb.colympus-lab.openstac.node2.cpu_pc	2	1494572996
stats.bb.colympus-lab.openstac.node2.fc_mb	0	1494572996
stats.bb.colympus-lab.openstac.node2.fc_io	0	1494572996
stats.bb.colympus-lab.openstac.node2.sas_mb	0	1494572996
stats.bb.colympus-lab.openstac.node2.sas_io	0	1494572996
stats.bb.colympus-lab.openstac.node2.iscsi_mb	0	1494572996
stats.bb.colympus-lab.openstac.node2.iscsi_io	0	1494572996
stats.bb.colympus-lab.openstac.node2.write_cache_pc	0	1494572996
stats.bb.colympus-lab.openstac.node2.total_cache_pc	0	1494572996
stats.bb.colympus-lab.openstac.node2.vdisk_mb	0	1494572996
stats.bb.colympus-lab.openstac.node2.vdisk_io	0	1494572996
stats.bb.colympus-lab.openstac.node2.vdisk_ms	0	1494572996
stats.bb.colympus-lab.openstac.node2.mdisk_mb	0	1494572996
stats.bb.colympus-lab.openstac.node2.mdisk_io	0	1494572996
stats.bb.colympus-lab.openstac.node2.mdisk_ms	0	1494572996
stats.bb.colympus-lab.openstac.node2.drive_mb	0	1494572996
stats.bb.colympus-lab.openstac.node2.drive_io	0	1494572996
stats.bb.colympus-lab.openstac.node2.drive_ms	0	1494572996
stats.bb.colympus-lab.openstac.node2.vdisk_r_mb	0	1494572996
stats.bb.colympus-lab.openstac.node2.vdisk_r_io	0	1494572996
stats.bb.colympus-lab.openstac.node2.vdisk_r_ms	0	1494572996
stats.bb.colympus-lab.openstac.node2.vdisk_w_mb	0	1494572996
stats.bb.colympus-lab.openstac.node2.vdisk_w_io	0	1494572996
stats.bb.colympus-lab.openstac.node2.vdisk_w_ms	0	1494572996
stats.bb.colympus-lab.openstac.node2.mdisk_r_mb	0	1494572996
stats.bb.colympus-lab.openstac.node2.mdisk_r_io	0	1494572996
stats.bb.colympus-lab.openstac.node2.mdisk_r_ms	0	1494572996
stats.bb.colympus-lab.openstac.node2.mdisk_w_mb	0	1494572996
stats.bb.colympus-lab.openstac.node2.mdisk_w_io	0	1494572996
stats.bb.colympus-lab.openstac.node2.mdisk_w_ms	0	1494572996
stats.bb.colympus-lab.openstac.node2.drive_r_mb	0	1494572996
stats.bb.colympus-lab.openstac.node2.drive_r_io	0	1494572996
stats.bb.colympus-lab.openstac.node2.drive_r_ms	0	1494572996
stats.bb.colympus-lab.openstac.node2.drive_w_mb	0	1494572996
stats.bb.colympus-lab.openstac.node2.drive_w_io	0	1494572996
stats.bb.colympus-lab.openstac.node2.drive_w_ms	0	1494572996
stats.bb.colympus-lab.openstac.node2.iplink_mb	0	1494572996
stats.bb.colympus-lab.openstac.node2.iplink_io	0	1494572996
stats.bb.colympus-lab.openstac.node2.iplink_comp_mb	0	1494572996
```

### CODE REVIEW

**Comment by knandya** [comment]
Let us mark the PR as WIP while we are figuring out the details.

File: sensu/plugins/check_v7k_canister.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    parser.add_argument('-p', '--v7k-port',
+                        help="ssh port for v7k")
+
+    parser.add_argument('-u', '--user',
+                        help="user to access v7k ")
+
+    parser.add_argument('-i', '--ssh-key',
+                        help="private key to access v7k")
+
+    args = parser.parse_args()
```
Comment (Xiaohua-Shen) [inline sensu/plugins/check_v7k_canister.py]: Need to add argument of '--criticality'

File: sensu/plugins/metrics_v7k.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    parser.add_argument('-p', '--v7k-port',
+                        help="ssh port for v7k")
+
+    parser.add_argument('-u', '--user',
+                        help="user to access v7k ")
+
+    parser.add_argument('-i', '--ssh-key',
+                        help="private key to access v7k")
+
+    args = parser.parse_args()
```
Comment (Xiaohua-Shen) [inline sensu/plugins/metrics_v7k.py]: Need to add argument of '--scheme' to avoid metrics mixture with other envs.

**Review by Xiaohua-Shen — CHANGES_REQUESTED** [review]

**Comment by nirajdp76** [comment]
I think checks need better exception handling. Also, would like to see sample outputs that operators will see when we raise critical alert.

**Review by nirajdp76 — CHANGES_REQUESTED** [review]
see comment

**Review by nirajdp76 — APPROVED** [review]

**Pull request merged**